biodatasets/twadrl/twadrl.py

```diff
@@ -0,0 +1,167 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import datasets
+import glob
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+
+_DATASETNAME = "twadrl"
+
+_CITATION = """
+@inproceedings{limsopatham-collier-2016-normalising,
+    title = "Normalising Medical Concepts in Social Media Texts by Learning Semantic Representation",
+    author = "Limsopatham, Nut  and
+      Collier, Nigel",
+    booktitle = "Proceedings of the 54th Annual Meeting of the Association for Computational Linguistics (Volume 1: Long Papers)",
+    month = aug,
+    year = "2016",
+    address = "Berlin, Germany",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/P16-1096",
+    doi = "10.18653/v1/P16-1096",
+    pages = "1014--1023",
+}
+"""
+
+_DESCRIPTION = """
+The TwADR-L dataset contains medical concepts written on social media (Twitter) \
+mapped to how they are formally written in medical ontologies (SNOMED-CT and AMT). \
+"""
+
+_HOMEPAGE = "https://zenodo.org/record/55013"
+
+_LICENSE = "CC BY 4.0"
+
+_URLs = "https://zenodo.org/record/55013/files/datasets.zip"
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.NAMED_ENTITY_DISAMBIGUATION]
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+
+class TwADRL(datasets.GeneratorBasedBuilder):
+    """TwADR-L: Dataset for Normalising Medical Concepts on Twitter."""
+
+    DEFAULT_CONFIG_NAME = "twadrl_source"
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="twadrl_source",
+            version=SOURCE_VERSION,
+            description="TwADR-L source schema",
+            schema="source",
+            subset_id="twadrl",
+        ),
+        BigBioConfig(
+            name="twadrl_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="TwADR-L simplified BigBio schema",
+            schema="bigbio_kb",
+            subset_id="twadrl",
+        ),
+    ]
+
+    def _info(self):
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "fold": datasets.Value("int32"),
+                    "cui": datasets.Value("string"),
```

Comment:
minor, but wouldn't an `int` be better?

Reply:
Just to double check, "int" instead of "int32" for `fold`? Or for `cui`? 

If the former, I'm not familiar with how integer size is handled in datasets/pyarrow (I'm guessing it'll default to 32?) so thought I should specify it. Maybe "int8" is more appropriate for number of folds? Would be happy to change this.

If the latter, the CUIs are strings here are always prefixed with "C".

Reply:
> If the latter, the CUIs are strings here are always prefixed with "C".

Argh! Sorry I got it mixed up with the other dataset. for "cui" string is fine. For "fold" "int32" is fine. Sorry for the confusion.